### PR TITLE
Correct day angle in ASCE extraterrestrial irradiance model

### LINKED ISSUE #211: Bug in extraterrestrial radiation calculation
The ET radiation calculator needs the "day angle" of the Earth's position in its orbit around the Sun. Our extraradiation function and the matlab pvl_extraradiation calculate the angle as:

```
B = (2. * pi / 365.) * doy
```

Assuming that doy is a value between 1 and 365, I believe it should be

```
B = (2. * pi / 365.) * (doy - 1)
```

The disc and pvl_disc functions use the (doy-1) formula.

### PR DESCRIPTION
**Pull request by cwhanse**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python! You may delete all of these instructions except for the list below.

You may submit a pull request with your code at any stage of completion.

The following items must be addressed before the code can be merged. Please don't hesitate to ask for help if you're unsure of how to accomplish any of the items below:

 - [x] Closes  #211 
 - [x] I am familiar with the [contributing guidelines](http://pvlib-python.readthedocs.io/en/latest/contributing.html).
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests (usually) must pass on the TravisCI and Appveyor testing services.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.
 - [x] Code quality and style is sufficient. Passes LGTM and SticklerCI checks.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

Change day angle equation to 2 pi / 365 * doy, where doy = 1, …, 365

### CODE REVIEW

File: pvlib/irradiance.py [hunk 1]
```diff
@@ -88,7 +92,7 @@ def get_extra_radiation(datetime_or_doy, solar_constant=1366.1,
     # consider putting asce and spencer methods in their own functions
     method = method.lower()
     if method == 'asce':
-        B = solarposition._calculate_simple_day_angle(to_doy(datetime_or_doy))
+        B = (2. * np.pi / 365.) * (to_doy(datetime_or_doy))
```
Comment (wholmgren) [inline pvlib/irradiance.py]: Alternative implementation: add an `offset` kwarg to `solarposition._calculate_simple_day_angle`:

```python
def _calculate_simple_day_angle(dayofyear, offset=1):
    """
    Calculates the day angle for the Earth's orbit around the Sun.

    Parameters
    ----------
    dayofyear : numeric
    offset : int, default 1
        choose 0 for the ASCE method or 1 for the Spencer method

    Returns
    -------
    day_angle : numeric
    """
    return (2. * np.pi / 365.) * (dayofyear - offset)
```

I'm fine with either approach

Comment (cwhanse) [reply pvlib/irradiance.py]: I like your way better

**Comment by wholmgren** [comment]
Ah, the test fails to pass the `method` kwarg to the function. Lots of failures when fixed.

**Comment by cwhanse** [comment]
Thanks @Schmelzersolar for investigating and reporting the issue.

**Pull request merged**